src/components/Chart/Chart.js

```diff
@@ -0,0 +1,297 @@
+                // every Chart will contain all scenarios and all severities
+                // startIdx and endIdx specify the time range on which we want to calc quantiles
+                // every sim.vals array will be sliced on this timeRange 
+                // then every day of a simulation will be summed up returning an Array of sim sums 
+                // then d3.quantiles can be applied to the Array to create final desired obj
+                
+                const sumArray = dataset[scenario][severity][stat].sims.map(sim => {
+                    return sim.vals.slice(startIdx, endIdx).reduce((a, b) => a + b, 0)
+                } );
+                // console.log(scenario, stat, severity, sumArray)
+                const minVal = min(sumArray)
+                const maxVal = max(sumArray)
+                const tenth = quantile(sumArray, 0.10)
+                const quartile = quantile(sumArray, 0.25)
+                const median = quantile(sumArray, 0.5)
+                const thirdquartile = quantile(sumArray, 0.75)
+                const ninetyith = quantile(sumArray, 0.9)
+                
+                // keep track of largest value across all the severities and scenarios for yAxis
+                if (maxVal > globalMaxVal) globalMaxVal = maxVal
+
+                quantileObj[stat][severity][scenario] = {
+                    minVal, maxVal, tenth, quartile, median, thirdquartile, ninetyith, //xScale, yScale
+                }
+            }
+        }
+        
+        console.log(quantileObj)
+        let yScale;
+        if (this.props.scale === 'linear') {
+            yScale = scaleLinear().range([height - margin.bottom, margin.top]).domain([0, globalMaxVal])
+        } else {
+           yScale = scalePow().exponent(0.25).range([height - margin.bottom, margin.chartTop]).domain([0, globalMaxVal])
+        }
+        // const yScale = scaleLog().range([height - margin.bottom, margin.top]).domain([1, globalMaxVal]) //
+        const xScale = scaleBand().range([0, (width / severities.length) - margin.left]).domain(scenarios)//.paddingInner(1).paddingOuter(.5);
+        this.setState({ quantileObj, xScale, yScale, scaleDomains: true })
+    }
+
+    handleHighlightEnter = (event, severity, key, index) => {
+        if (!this.state.rectIsHovered) {
+            console.log('handleHighlightEnter', this.state.rectIsHovered)
+            event.stopPropagation();
+            // console.log(severity, key, index)
+            const hoveredRect = {
+                'severity': severity,
+                'scenario': key,
+                'index': index
+            }
+            const { severities, quantileObj }  = this.state;
+            const { stat, statLabel, summaryStart, summaryEnd, width } = this.props;
+            // const formatDate = timeFormat('%b %d, %Y'); //timeFormat('%Y-%m-%d')
+            const median = quantileObj[stat][severity][key]['median']
```

Comment:
You might want to use a typechecker (typescript).

This will let you do things like `quantileObj.stat.severity.key.median`. This allows the IDE to autocomplete keys as well as validate that it does/doesn't exist without having to run the code.